### DISCUSSION TITLE
Specifying "desc" in forecast vs. obs in PointStat

### DISCUSSION THREAD

**Discussion by gsketefian** [dtcenter/METplus#2445] [Configuration]
Hi,

I have a METplus config file for PointStat that has this section:
```
FCST_VAR10_NAME = HPBL
FCST_VAR10_LEVELS = Z0
FCST_VAR10_OPTIONS = set_attr_lead = "{lead?fmt=%H%M%S}";
                     desc = "TKE";
OBS_VAR10_NAME = PBL
OBS_VAR10_LEVELS = L0

FCST_VAR11_NAME = HGT
FCST_VAR11_LEVELS = L0
FCST_VAR11_OPTIONS = set_attr_lead = "{lead?fmt=%H%M%S}";
                     GRIB_lvl_typ = 220;
                     desc = "RI";
OBS_VAR11_NAME = PBL
OBS_VAR11_LEVELS = L0
```
Here, the obs for VAR10 and VAR11 are the same; they're just the PBL height.  But for the forecast, VAR10 is the PBL height calculated using the TKE, and VAR11 is the PBL height using the Richardson number (RI).  So it makes sense that the `desc = "[TKE|RI]"` options are set for the forecast but not the obs.  The issue I have is that the resulting stat file contains `NA` in the `DESC` column of these variables instead of `TKE` or `RI`.  But if I switch the `desc = ...` options to the obs, i.e. 
```
FCST_VAR10_NAME = HPBL
FCST_VAR10_LEVELS = Z0
FCST_VAR10_OPTIONS = set_attr_lead = "{lead?fmt=%H%M%S}";
OBS_VAR10_NAME = PBL
OBS_VAR10_LEVELS = L0
OBS_VAR10_OPTIONS = desc = "TKE";

FCST_VAR11_NAME = HGT
FCST_VAR11_LEVELS = L0
FCST_VAR11_OPTIONS = set_attr_lead = "{lead?fmt=%H%M%S}";
                     GRIB_lvl_typ = 220;
OBS_VAR11_NAME = PBL
OBS_VAR11_LEVELS = L0
OBS_VAR11_OPTIONS = desc = "RI";
```
then `TKE` and `RI` do show up in the `DESC` column of the stat file.  But this is misleading because it makes it look like the obs for the two variables are somehow different, and it doesn't clearly indicate that the forecasts are the ones that are different.  Is there a way to make the `desc` option for the FCST variables show up in the stat file?

Thanks,
Gerard

**Comment by JohnHalleyGotway** [comment]
@gsketefian I see you have a question about setting the description column (**DESC**) in the MET output. The short answer is no, specifying `desc = "TKE";` in the ** FCST_VAR{N}_OPTIONS** setting won't have the desired effect, as you've already found.

Here's a link to the relevant documentation in the [MET User's Guide](https://met.readthedocs.io/en/latest/Users_Guide/config_options.html#desc):
`It may be set separately in each “obs.field” verification task entry or simply once at the top level of the configuration file.`

There's several options like this that can be specified separately for each verification task. Look in the [PointStatConfig_default](https://github.com/dtcenter/MET/blob/main_v11.1/data/config/PointStatConfig_default) file for entries with `May be set separately in each "obs.field" entry`.

Logically, as you note, these things could be specified in the forecast dictionary or the observation dictionary. For simplicity, we chose the convention noted above, that settings in this category should be specified within the observation dictionary.

I can see how the code could be enhanced to support what you're requesting, but it isn't a trivial change. Because we always read the default config file, the default `desc = "NA";` setting will always exist at the highest level of config file context. We could make the parsing code pay more careful attention to the context from which these entries are parsed. It would need to attempt to parse each from both the forecast and observation entries. If found in neither, look to the higher level of config context. If found in one but not the other, use that value. If found in both, use it if they match, and error out if not. Rather than implementing that logic initially, we chose to adopt the simpler convention of parsing them from the obs dictionary instead.

In this specific case, I'd actually recommend that you consider using the `set_attr_name` option instead:
```
FCST_VAR10_NAME = HPBL
FCST_VAR10_LEVELS = Z0
FCST_VAR10_OPTIONS = set_attr_lead = "{lead?fmt=%H%M%S}"; set_attr_name = "PBL_TKE";

FCST_VAR11_NAME = HGT
FCST_VAR11_LEVELS = L0
FCST_VAR11_OPTIONS = set_attr_lead = "{lead?fmt=%H%M%S}";
                     GRIB_lvl_typ = 220; set_attr_name = "PBL_RI";
```
That would set the `FCST_VAR` column in the output .stat files as `PBL_TKE` and `PBL_RI` directly instead of stashing that info in the `DESC` column. Personally, I find that solution more intuitive, but I understand if you'd prefer to denote it in the `DESC` column instead.

So I'll pose a question to you. As a METplus user, do you think it's worth the effort to document this feature request with a new GitHub issue in the MET repo? Would the flexibility of specifying the description value in the forecast *OR* observation settings greatly improve the usability/configurability of MET? If so, I'm happy to add it and confident that a good solution exists. We would also need to update the documentation and config file contents to state `May be set separately in each "obs.field" or "fcst.field" entry`.
